BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -51,7 +51,12 @@ public class CryptoStore: ObservableObject {
       }
     }
   }
-  @Published private(set) var pendingRequest: PendingRequest?
+  @Published private(set) var pendingRequest: PendingRequest? {
+    didSet {
+      pendingRequestUpdated?()
+    }
+  }
+  private let pendingRequestUpdated: (() -> Void)?
```

Comment:
Why store a closure to call when it updates? Make this `public private(set)` and subscribe to updates via Combine's `sink`?

Reply:
The `pendingRequest` was kept internal so we don't have to change the `PendingRequest` enum to public when the we only care about here checking when it updates (on update we are calling the `isPendingRequestAvailable()` function not checking the `pendingRequest: PendingRequest?` which can be nil _with_ a valid pending request when user dismisses a request without accepting/rejecting). I can use a sink on it internally and call this closure but I'd still need to store the `AnyCancellable` here.

Reply:
Addressed by using a `PassthroughSubject` instead of injecting a closure here [d5e0825](https://github.com/brave/brave-ios/pull/5383/commits/d5e08253188a8b9657ffff2e8003a1ec38824fbd)